source/common/http/header_utility.cc

```diff
@@ -164,5 +170,17 @@ bool HeaderUtility::isEnvoyInternalRequest(const HeaderMap& headers) {
          internal_request_header->value() == Headers::get().EnvoyInternalRequestValues.True;
 }
 
+bool HeaderUtility::requestHeadersValid(const HeaderMap& headers, const std::string** details) {
```

Comment:
nit: you could possibly simplify this by just returning an `absl::optional<const std::string*>` or `absl::optional<std::reference_wrapper<const std::string>>` and avoid the second parameter, but up to you.